compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -160,6 +160,14 @@ impl<'ll, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'_, 'll, 'tcx> {
                     _ => bug!("the va_arg intrinsic does not work with non-scalar types"),
                 }
             }
+            // FIXME: why is this needed when `type_id` is not?
```

Comment:
Definitely very surprising.  The things in codegen_ssa are attempted first, which you've updated, so I agree this shouldn't be needed.